lib/CXGN/Trial/TrialCreate.pm

```diff
@@ -146,13 +146,14 @@ has 'dbh' => (is  => 'rw',predicate => 'has_dbh', required => 1,);
 #has 'user_name' => (isa => 'Str', is => 'rw', predicate => 'has_user_name', required => 1,);
 has 'trial_id' => (isa => 'Maybe[Int]', is => 'rw', predicate => 'has_trial_id');
 has 'program' => (isa =>'Str', is => 'rw', predicate => 'has_program', required => 1,);
-has 'trial_year' => (isa => 'Str', is => 'rw', predicate => 'has_trial_year', required => 1,);
-has 'trial_description' => (isa => 'Str', is => 'rw', predicate => 'has_trial_description', required => 1,);
-has 'trial_location' => (isa => 'Str', is => 'rw', predicate => 'has_trial_location', required => 1,);
-has 'design_type' => (isa => 'Str', is => 'rw', predicate => 'has_design_type', required => 1);
-has 'design' => (isa => 'HashRef[HashRef]|Undef', is => 'rw', predicate => 'has_design', required => 1);
+has 'trial_year' => (isa => 'Maybe[Str]', is => 'rw', predicate => 'has_trial_year', required => 0,);
+has 'trial_description' => (isa => 'Maybe[Str]', is => 'rw', predicate => 'has_trial_description', required => 0,);
+has 'trial_location' => (isa => 'Maybe[Str]', is => 'rw', predicate => 'has_trial_location', required => 0,);
+has 'design_type' => (isa => 'Maybe[Str]', is => 'rw', predicate => 'has_design_type', required => 0);
+has 'design' => (isa => 'HashRef[HashRef]|Undef', is => 'rw', predicate => 'has_design', required => 0);
 has 'trial_name' => (isa => 'Str', is => 'rw', predicate => 'has_trial_name', required => 1);
-has 'trial_type' => (isa => 'Str', is => 'rw', predicate => 'has_trial_type', required => 0);
+has 'trial_type' => (isa => 'Str', is => 'rw', predicate => 'has_trial_type', required => 1);
+has 'trial_type_value' => (isa => 'Str', is => 'rw', predicate => 'has_trial_type', required => 0);
```

Comment:
This does not compile because there are now 2 has_trial_type predicates

Reply:
Good catch! I'll fix that today. I am also going to remove the optional design code and make it required again. It seems too central to other BreedBase features to not have a design code.

Reply:
Predicate has been fixed. I made design type required again (`experimentalDesign.PUI` in brapi).